docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -317,16 +335,106 @@ To enable this, we simply need to add `Contracts` type that we defined in the `c
+We now need to enable calls to the contracts pallet. Not all of the pallets require using a specialized configuration to interact with them. However, the contracts pallet requires these API to call the sandboxed contracts.
+
+To achieve this, we need to start by adding the required API dependencies.
+
+**`runtime/Cargo.toml`**
+```TOML
+[dependencies.contracts-rpc-runtime-api]
+default-features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'pallet-contracts-rpc-runtime-api'
+rev = '<git-commit>' # e.g. '52373bfe63d49aae7a17b10b17116a3d470d30bf'
+```
+
+**`runtime/Cargo.toml`**
+```TOML
+[features]
+default = ["std"]
+std = [
+    #--snip--
+    'contracts-rpc-runtime-api/std',
+]
+```
+
+Contracts do not return data at the end of their execution. Due to the nature of blockchains, a contract execution needs to be valid to get finalized and included in a block before getting the current state of any variable in it.
+
+To get the state of a variable, we have to call a getter function that will return a `ContractExecResult` wrapper with the current state of the execution.
+
+To use this return type, we need to add it to our runtime.
+
+**`runtime/src/lib.rs`**
+```rust
+/* --snip-- */
+use sp_std::prelude::*;
+
+/*** Add this line ***/
+use contracts_rpc_runtime_api::ContractExecResult;
+/* --snip-- */
+```
+
+The next step, is to implement the required functions to handle the returned data.
+
+```rust
+impl_runtime_apis! {
+   /* --snip-- */
+
+   /*** Add this block ***/
+    impl contracts_rpc_runtime_api::ContractsApi<Block, AccountId, Balance> for Runtime {
+        fn call(
+            origin: AccountId,
+            dest: AccountId,
+            value: Balance,
+            gas_limit: u64,
+            input_data: Vec<u8>,
+        ) -> ContractExecResult {
+            let exec_result = Contracts::bare_call(
+                origin,
+                dest.into(),
+                value,
+                gas_limit,
+                input_data,
+            );
+            match exec_result {
+                Ok(v) => ContractExecResult::Success {
+                    status: v.status,
+                    data: v.data,
+                },
+                Err(_) => ContractExecResult::Error,
+            }
+        }
+
+        fn get_storage(
+            address: AccountId,
+            key: [u8; 32],
+        ) -> contracts_rpc_runtime_api::GetStorageResult {
+            Contracts::get_storage(address, key).map_err(|rpc_err| {
+                use contracts::GetStorageError;
+                use contracts_rpc_runtime_api::{GetStorageError as RpcGetStorageError};
+                /// Map the contract error into the RPC layer error.
+                match rpc_err {
+                    GetStorageError::ContractDoesntExist => RpcGetStorageError::ContractDoesntExist,
+                    GetStorageError::IsTombstone => RpcGetStorageError::IsTombstone,
+                }
+            })
+        }
+    }
+   /*** ***/
```

Comment:
What is this empty comment for?

Reply:
I'm trying to use it as a way to indicate the end of a block. It has the same pattern as the `/*** Add this block ***/`. If it is not clear I can change/remove it.

Reply:
I see what you mean now. Maybe consider `/*** End Added Block ***/`?? 

I'll leave it up to you.